src/main/java/org/opennars/inference/RuleTables.java

```diff
@@ -56,7 +56,12 @@ public static void reason(final TaskLink tLink, final TermLink bLink, final Deri
         
         final Concept beliefConcept = memory.concept(beliefTerm);
```

Comment:
Here the belief concept might reference a old deleted one - no idea if locking here leads to a deadlock

Reply:
This is probably fine for now because it is unlikely